Sources/Swift/Helper/SentryCurrentDateProvider.swift

```diff
@@ -1,16 +1,24 @@
 @_implementationOnly import _SentryPrivate
 import Foundation
 
-@objcMembers class SentryCurrentDateProvider: NSObject {
-    
+@objc
+protocol SentryCurrentDateProvider {
+    func date() -> Date
+    func timezoneOffset() -> Int
+    func systemTime() -> UInt64
+    func systemUptime() -> TimeInterval
+}
```

Comment:
💯 way better than the other solution and the refactoring isn't that big either:
![CleanShot 2024-10-15 at 14 00 15@2x](https://github.com/user-attachments/assets/dd75d8c1-339a-4b0c-854c-ecd1cc8f5c6d)